tmt/steps/prepare/artifact/providers/koji.py

```diff
@@ -179,11 +216,15 @@ def _call_api(self, method: str, *args: Any, **kwargs: Any) -> Any:
         except Exception as error:
             raise tmt.utils.GeneralError(f"API call '{method}' failed.") from error
 
-    @cached_property
-    def build_provider(self) -> Optional['KojiBuild']:
+    def _make_build_provider(self, build_cls: type[TBuild], prefix: str) -> Optional[TBuild]:
+        """Create a build provider instance if build_id is available."""
         if self.build_id is None:
             return None
-        return KojiBuild(f"koji.build:{self.build_id}", self.logger)
+        return build_cls(f"{prefix}{self.build_id}", self.logger)
+
+    @cached_property
+    def build_provider(self) -> Optional['KojiBuild']:
+        return self._make_build_provider(KojiBuild, "koji.build:")
```

Comment:
I would recommend not making `:` part of the "prefix" argument. Every caller of `_make_build_provider` would need to include it, they would need to be aware of `:` being a separator. It's easy to add it inside the f-string in `_make_build_provider`. Seems to apply to `_REGISTRY` as well.